api/utils.py

```diff
@@ -127,220 +128,299 @@ async def __call__(self, request: Request, security_scopes: SecurityScopes, retu
+        ):
+            if self.custom_methods.get("get"):
+                return await self.custom_methods["get"](pagination, user)
+            else:
+                return await pagination.paginate(self.orm_model, user.id)
+
+        return get
+
+    def _get_count(self):
+        async def get_count(
+            user: Union[None, ModelView.schemes.User] = Security(self.auth_dependency, scopes=self.scopes["get_count"])
+        ):
+            return (
+                await (
+                    (
+                        self.orm_model.query.where(self.orm_model.user_id == user.id)
+                        if self.orm_model != models.User
+                        else self.orm_model.query
+                    )
+                    .with_only_columns([db.db.func.count(distinct(self.orm_model.id))])
+                    .order_by(None)
+                    .gino.scalar()
+                )
+                or 0
             )
-            or 0
-        )
 
-    async def get_one(model_id: int, request: Request):
-        try:
-            user = await auth_dependency(request, SecurityScopes(scopes["get_one"]))
-        except HTTPException:
-            if get_one_auth:
-                raise
-            user = None
-        return await _get_one(model_id, user)
+        return get_count
 
-    async def post(
-        model: create_model,
-        request: Request,  # type: ignore,
-    ):
+    async def get_one(self, model_id: int, request: Request):
         try:
-            user = await auth_dependency(request, SecurityScopes(scopes["post"]))
+            user = await self.auth_dependency(request, SecurityScopes(self.scopes["get_one"]))
         except HTTPException:
-            if post_auth:
+            if self.get_one_auth:
                 raise
             user = None
-        try:
-            if custom_methods.get("post"):
-                obj = await custom_methods["post"](model, user)
-            else:
-                obj = await orm_model.create(**model.dict())  # type: ignore
-        except (
-            asyncpg.exceptions.UniqueViolationError,
-            asyncpg.exceptions.NotNullViolationError,
-            asyncpg.exceptions.ForeignKeyViolationError,
-        ) as e:
-            raise HTTPException(422, e.message)
-        if background_tasks_mapping.get("post"):
-            background_tasks_mapping["post"].send(obj.id)
-        return obj
-
-    async def put(
-        model_id: int,
-        model: pydantic_model,
-        user: Union[None, schemes.User] = Security(auth_dependency, scopes=scopes["put"]),
-    ):  # type: ignore
-        item = await _get_one(model_id, user, True)
-        try:
-            if custom_methods.get("put"):
-                await custom_methods["put"](item, model, user)  # pragma: no cover
+        return await self._get_one(model_id, user)
+
+    def _post(self):
+        model_type = self.create_model
```

Comment:
Have to create a function wrapper to make model_type lazy-loaded.

Reply:
From investigating, indeed, function wrapper seems to be the most clear solution.
But is creating `model_type` variable necessary here?